tests/azure/test_azure.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+import os
+import time
+
+from azure.storage.blob import BlobServiceClient
+from azure.core.exceptions import ServiceRequestError, ResourceNotFoundError
+
+from tests.cases.interface import ObjectStoreBackend
+from tests.cases.testcases import compatibility_test_cases
+
+
+class AzuriteServer(ObjectStoreBackend):
+    """A helper class to interact with Azurite (Azure Storage emulator) test server."""
+    def __init__(self):
+        self.connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+        self.client = BlobServiceClient.from_connection_string(self.connection_string)
+
+    def wait_for_startup(self, timeout=30):
+        """Wait for the Azurite server to become available."""
+        print("Waiting for Azurite server to be up and running.")
+        start_time = time.time()
+        while time.time() - start_time < timeout:
+            try:
+                # Try to list containers to check connectivity
+                containers = self.client.list_containers(results_per_page=1)
+                next(iter(containers), None)  # Consume one item from the iterator
+                print("Azurite server is up and running.")
+                return
+            except (ServiceRequestError, ResourceNotFoundError, StopIteration):
+                time.sleep(0.5)
+        raise TimeoutError(f"Azurite server failed to start within {timeout} seconds.")
+
+    def upload_file(self, container_name, directory, file_path):
+        """Uploads a local file to the specified Azure container and directory."""
+        container_client = self.client.get_container_client(container_name)
+        
+        # Create container if it doesn't exist
+        try:
+            container_client.create_container()
+        except Exception:
```

Comment:
Is there another way to check if container exists?
And let real exceptions be raised further